Bottom: 1877637c3f07dc7e336aa78a4242e67d5af51ddf
Top:    e5a2323699818c850ce90a611481d160010b7fcb
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-29 16:17:52 -0500

Refresh of fixup-asset-host-2

---
diff --git a/roles/buildbot-master/files/openembedded.py b/roles/buildbot-master/files/openembedded.py
index 5c2d9da5..89c57a2c 100755
--- a/roles/buildbot-master/files/openembedded.py
+++ b/roles/buildbot-master/files/openembedded.py
@@ -66,13 +66,20 @@ c['schedulers'] = [
             util.BooleanParameter(
                 name="clobber",
                 label="Clobber build directory",
-                default=False
-            ),
+                default=False),
             util.BooleanParameter(
                 name='cache',
                 label="Use cached state",
-                default=True
-            ),
+                default=True),
+            util.StringParameter(
+                name="release_pin",
+                label="PIN for release signing",
+                default='',
+                required=False),
+            util.StringParameter(
+                name='bbflags',
+                label="BitBake Options",
+                default=DEFAULT_BBFLAGS),
             util.StringParameter(
                 name="release_pin",
                 label="PIN for release signing",
@@ -140,6 +147,17 @@ def ComputeBuildProperties(props):
     newprops['timestamp'] = CurrentTime()
 
     version = props.getProperty('version', default=newprops['timestamp'])
+
+    urlpath = 'ntel/unofficial'
+
+    if props.getProperty('release_pin') is True:
+        urlpath = 'ntel/release'
+
+    if props.getProperty('scheduler') == 'ntel-nightly':
+        urlpath = 'ntel/nightly'
+
+    urlpath = os.path.join('ntel', urlpath, version)
+
     newprops['artifacts'] = {}
     machines = [props.getProperty('machine')]
 
@@ -253,9 +271,9 @@ class BitBakeArchive(steps.ShellSequence):
             'description': 'archiving',
             'descriptionDone': 'archive',
             'env': {
-                    'ENV': 'environment-ntel',
-                    'BASH_ENV': 'environment-ntel',
-                    },
+                'ENV': 'environment-ntel',
+                'BASH_ENV': 'environment-ntel',
+            },
             'flunkOnFailure': True,
             'haltOnFailure': False,
             'name': 'archive',
@@ -283,6 +301,18 @@ class BitBakeArchive(steps.ShellSequence):
                 ],
                 logfile="Create %s archive" % (m),
             ))
+
+            arturl = art['url']
+            commands.append(util.ShellArg(
+                logfile="Upload %s archive" % (m),
+                haltOnFailure=False,
+                flunkOnFailure=True,
+                command=[
+                    'curl', '--netrc', '--verbose',
+                    '--upload-file', artfile,
+                    '--url', arturl
+                ],
+            ))
         return steps.ShellSequence.runShellSequence(self, commands)
